CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/LayersViewController.java

```diff
@@ -86,13 +96,116 @@ public void submit() {
         }
 
         final List<String> layerQueries = new ArrayList<>();
-        for (Node node : pane.getLayers().getChildren()) {
+        for (final Node node : pane.getLayers().getChildren()) {
             if (GridPane.getRowIndex(node) > 0 && GridPane.getColumnIndex(node) == 2) {
                 final TextArea queryTextArea = (TextArea) node;
                 layerQueries.add(queryTextArea.getText().isBlank() ? null : queryTextArea.getText());
             }
         }
-
         PluginExecution.withPlugin(new UpdateGraphQueriesPlugin(layerQueries)).executeLater(GraphManager.getDefault().getActiveGraph());
     }
+
+    /**
+     * Executes a plugin to grab current Layers View selections and save them to
+     * the graph's Layers View State Attribute.
+     */
+    public void updateState() {
+        pane = parent.getContent();
+        PluginExecution.withPlugin(new LayersViewStateUpdater(pane, true)).executeLater(GraphManager.getDefault().getActiveGraph());
+    }
+
+    /**
+     * Executes a plugin to grab current Layers View state and load them into
+     * the Layers View UI.
+     */
+    public void loadOrCreateState() {
+        pane = parent.getContent();
+        PluginExecution.withPlugin(new LayersViewStateUpdater(pane, false)).executeLater(GraphManager.getDefault().getActiveGraph());
+    }
+
+    /**
+     * Update the display by reading and writing to/from the state attribute.
+     */
+    private static final class LayersViewStateUpdater extends SimpleEditPlugin {
+
+        private LayersViewPane pane = null;
+        private boolean isUpdateCall = false;
+
+        public LayersViewStateUpdater(final LayersViewPane pane, final boolean isUpdateCall) {
+            this.pane = pane;
```

Comment:
You don't need to store `pane` within the plugin here - you should just be able to call `parent.getContent()` in the `captureState` method. That being said, rather than processing the pane within the plugin at all, I think it would make more sense for this class to offer `getLayers` and `setLayers` methods which operate on a list of `LayerDescription` objects. That way you could break the plugin away from this class (and add it to a registry).

Reply:
I tried to call parent.getContent(); within the plugin but it would not allow me because of the static context of the plugin. I'm in the process of cleaning a heap up - thanks to your suggestions it should be a load simpler.

Reply:
@aldebaran30701 so should I wait before reviewing this PR again?

Reply:
Yes, I will ping you both when the fixes are all pushed up. 👍